src/visitpy/common/Logging.C

```diff
@@ -1640,19 +1649,112 @@ static std::string log_ResetPickAttributesRPC(ViewerRPC *rpc)
 
 static std::string log_AddAnnotationObjectRPC(ViewerRPC *rpc)
 {
-    return MESSAGE_COMMENT("AddAnnotationObject", MSG_NOT_IMPLEMENTED);
+    int annotType = rpc->GetIntArg1();
+    // annotType may have a 'visible/not visible' bit.
+    // this next is taken from VisWinAnnotations::AddAnnotationObject
+    // in order to remove the visible bit and get at just the annotType
+    int const static CREATE_ANNOTATION_OBJECT_AS_NOT_VISIBLE = 0x00010000;
+    std::string visible("1");
+    if (!(0 <= annotType && annotType <= 8))
```

Comment:
what do the 0,8 values represent?

Reply:
The user-addable annotation types from:
https://github.com/visit-dav/visit/blob/39228d081ddbbdef7894ebcd9c1b500e521900ba/src/common/state/AnnotationObject.h#L31-L44

Reply:
why not change it to 

```c++
if (!(0 <= annotType && annotType < AnnotationType::MaxAnnotationType))
```

Then it is less brittle to changes to the enum. I think that's what the `MaxAnnotationType` is there for.

Reply:
Could change it to ` < AnnotationType::LegendAttributes`, but cannot use `MaxAnnotationType`, because Legends aren't a user-addable object, they are automatically added by a Plot, the if-test needs to consider only the first 8 items in the enum.

Reply:
I see. This is why I dislike enums; there's no good way to check if a value is a valid enum value. I'm not a huge fan of comparing by value, but comparing via `LegendAttributes` isn't great either in case someone someday adds a new annotation type after it in the enum that is user-addable. I'm at a loss.

Reply:
Could you define a `MaxUserManagedAnnotationType`? Or, could you create a `const enum AnnotationType MaxUserManagedAnnotationType = Image`?

Reply:
Will the changes I made to the comments satisfy?

Reply:
I guess so. Mark's suggestion makes the code less brittle to changes.

Reply:
> I guess so. Mark's suggestion makes the code less brittle to changes.

Only if Image is always that last user-addable object in the list.

Reply:
There's no way to win with enums. Enum classes are better. :(

Maybe this modification to Mark's suggestion would be good?
```
const enum AnnotationType MaxUserManagedAnnotationType = 
   static_cast<enumAnnotationType>(AnnotationType::MaxAnnotationType - 2);
```

But I think these nitpicks are not important. I went ahead and approved the PR.

Reply:
Here's a new thought, re Mark's suggestion, I could add 'MaxUserManagedAnnotationType' to the enum list, just before LegendAttributes, then use that as the comparison value when needed. Then, if we ever add new objects to the list, we wouldn't have to change logic elsewhere.